plugins/processors/topk/README.md

```diff
@@ -0,0 +1,97 @@
+# TopK Processor Plugin
+
+The TopK processor plugin is a filter designed to get the top series over a period of time. It can be tweaked to do its top k computation over a period of time, so spikes can be smoothed out.
+
+This plugin groups the metrics based on their name and tags, then generates aggregates values across each group base on fields selected by the user. It then sorts these groups based on these aggregations and returns any metric that belongs to a group in the top k (sorted by any of the aggregations). This means that when calculating the top k, more than k metrics may be returned.
+
+If only the very top k metrics are needed, regardless of grouping, the simple_topk setting will force each metric into its own individual group
+
+### Configuration:
+
+```toml
+[[processors.topk]]
+  ## How many seconds between aggregations
+  # period = 10
+
+  ## How many top metrics to return
+  # k = 10
+
+  ## Metrics are grouped based on their tags and name. The plugin aggregates
+  ## the selected fields of these groups of metrics and sorts the groups based
+  ## these aggregations
+
+  ## Over which tags should the aggregation be done. Globs can be specified, in
+  ## which case any tag matching the glob will aggregated over. If set to an
+  ## empty list is no aggregation over tags is done
+  # group_by = ['*']
+
+  ## Wheter or not to also group by metric name
+  # group_by_metric_name = true
+
+  ## The plugin can aggregate over several fields. If more than one field is
+  ## specified, an aggregation is calculated per group per field.
+
+  ## The plugin returns a metric if it's in a group in the top k groups,
+  ## ordered by any of the aggregations of the selected fields
+
+  ## This effectively means that more than K metrics may be returned. If you
+  ## need to return only the top k metrics regardless of grouping, use the simple_topk setting
+
+
+  ## Over which fields are the top k are calculated
+  # fields = ["value"]
+
+  ## What aggregation to use. Options: sum, mean, min, max
+  # aggregation = "mean"
+
+  ## Instead of the top k largest metrics, return the bottom k lowest metrics
+  # bottomk = false
+
+  ## If true, this will override any GroupBy options and assign each metric
+  ## its own individual group. Default: false
+  # simple_topk = false
+
+  ## Drop any metrics that do fit in any group (due to nonexistent tags)
+  # drop_no_group = true
+
+  ## Drop the metrics that do not make the cut for the top k
+  # drop_non_top = true
```

Comment:
I don't see how `k` implements this functionality. This is here to give the user the option *not to drop any metrics* (by setting this to false), which might seem like a dumb option in a plugin whose sole purpose is to keep a subset of the metrics, but it can be useful because this plugin can also *annotate* metrics.

This option is here for users that would like TopK to _annotate_ their metrics (as you can do with the rank and aggregate field options) but not drop them, letting them effectively do some of the dropping or any other post processing afterwards.

I do think though that that is a very edgy case, and that for the vast mayority of users this will remain as the default, so I'm not terribly attached to it. If you guys still want it gone, Ill delete the feature

Reply:
If `k` was set to a suitably large value then you would get all values, which is why I feel like the options are related.  In some contexts it is common for 0 to indicate infinity, such as when specifying a timeout for a network operation.  Either way is okay.

Reply:
Yeah, but a very large k would also **annotate** all values, which it's something you might not want.

Reply:
@danielnelson Still waiting on input on this comment. Going by our previous discussion regarding simple top k (which since then I've already removed) I think it makes sense to remove this option too. It's an instance of the plugin trying to do too many things.

Reply:
Yeah, I think I would remove the option, wanting to annotate some of the values seems like a fringe use.